### PR TITLE
Remove UA Parser from expo-constants

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

UA Parser is too big to be automatically bundled in every expo project.

Screen Shot 2020-01-20 at 1 22 40 PM

# How

- `expo-constants` `Constants.deviceName` now only returns the possible Browser name and doesn't fallback to engine or OS name.
- `expo-constants` `Constants.platform.web` now only returns the `ua` (user agent string).

### CODE REVIEW

File: packages/expo-constants/src/ExponentConstants.web.ts [hunk 1]
```diff
@@ -15,6 +13,30 @@ declare var localStorage: Storage;
 
 const _sessionId = uuidv4();
 
+function getBrowserName(): string | undefined {
+  const agent = navigator.userAgent.toLowerCase();
+  switch (true) {
```
Comment (ide) [inline packages/expo-constants/src/ExponentConstants.web.ts]: just write if/else

File: packages/expo-constants/src/ExponentConstants.web.ts [hunk 2]
```diff
@@ -15,6 +13,30 @@ declare var localStorage: Storage;
...
+    case agent.includes('edge'):
+      return 'Edge';
+    case agent.includes('edg'):
+      return 'Chromium Edge';
+    // @ts-ignore: window is not defined
```
Comment (ide) [inline packages/expo-constants/src/ExponentConstants.web.ts]: TS is set up to allow `window`. I just tested locally and VS Code didn't show any TS errors and `yarn tsc` succeeded. Remove these unnecessary `@ts-ignore` comments and fix your editor if it's complaining about `window` (but not `navigator`??)

**Review by ide — CHANGES_REQUESTED** [review]

File: packages/expo-constants/src/ExponentConstants.web.ts (hunk 2 above)
Comment (EvanBacon) [reply packages/expo-constants/src/ExponentConstants.web.ts]: https://app.circleci.com/jobs/github/expo/expo/119077/parallel-runs/0/steps/0-114

```
PASS iOS src/__tests__/Constants-test.ts (5.8s)
PASS Android src/__tests__/Constants-test.ts
PASS Web src/__tests__/Constants-test.ts
FAIL Node src/__tests__/Constants-test.ts
  ● Test suite failed to run

    ReferenceError: window is not defined

      15 | 
      16 | function getBrowserName(): string | undefined {
    > 17 |   const agent = window.navigator.userAgent.toLowerCase();
         |                 ^
      18 |   switch (true) {
      19 |     case agent.includes('edge'):
      20 |       return 'Edge';

      at getBrowserName (src/ExponentConstants.web.ts:17:17)
      at Object.get deviceName [as deviceName] (src/ExponentConstants.web.ts:90:12)
      at _objectWithoutPropertiesLoose (../../node_modules/@babel/runtime/helpers/objectWithoutPropertiesLoose.js:10:25)
      at _objectWithoutProperties (../../node_modules/@babel/runtime/helpers/objectWithoutProperties.js:5:16)
      at Object.<anonymous> (src/Constants.ts:48:55)
      at Object.<anonymous> (src/__tests__/Constants-test.ts:1:1)

Test Suites: 1 failed, 3 passed, 4 total
Tests:       6 passed, 6 total
Snapshots:   0 total
Time:        15.132s
Ran all test suites in 4 projects.
error Command failed with exit code 1.
```

Comment (ide) [reply packages/expo-constants/src/ExponentConstants.web.ts]: Gotcha, that makes sense. It's a bit sloppy but maybe we should add '.node' to web linter rules.

**Review by ide — APPROVED** [review]

**Review by bbarthec — APPROVED** [review]

**Pull request merged**